lib/membrane/video_compositor/scene/layout.ex

```diff
@@ -0,0 +1,42 @@
+defmodule Membrane.VideoCompositor.Scene.Layout do
+  @moduledoc """
+  Structure representing Layout objects.
+
+  Layouts can take multiple renderable objects as input and
+  combine them into one frame. Fadings, Grids, Overlays, Transitions,
+  etc. can be defined as Layouts.
+
+  Basically multi-input, single-output node in processing graph.
+  """
+  alias Membrane.VideoCompositor.Scene.Object
+
+  @typedoc """
+  Keys enforced in Layout objects.
+  """
+  @type enforced_keys :: :inputs | :resolution
+
+  @typedoc """
+  Specify :inputs types in Layout structs.
```

Comment:
I think it would be better to focus on semantics of that type rather than the fact that it's used in the `:inputs` map.